powerauth-restful-server-spring/src/main/java/io/getlime/security/powerauth/app/rest/api/spring/controller/v3/EncryptedDataExchangeController.java

```diff
@@ -176,17 +178,59 @@ public String exchangeSignedAndEncryptedDataString(@EncryptedRequestBody String
                                                                PowerAuthApiAuthentication auth) throws PowerAuthAuthenticationException, PowerAuthEncryptionException {
 
         if (auth == null || auth.getUserId() == null) {
-            logger.debug("Signature validation failed");
+            logger.error("Signature validation failed");
             throw new PowerAuthSignatureInvalidException();
         }
 
         if (eciesContext == null) {
-            logger.debug("Encryption failed");
+            logger.error("Encryption failed");
             throw new PowerAuthEncryptionException();
         }
 
         // Return data back for verification
         return requestData;
     }
 
+    /**
+     * Sample signed and encrypted data exchange of generified request.
+     *
+     * @param request Request with generified request data.
+     * @param eciesContext ECIES context.
+     * @param auth PowerAuth authentication object.
+     * @return Generified data exchange response.
+     * @throws PowerAuthAuthenticationException In case signature validation fails.
+     * @throws PowerAuthEncryptionException In case encryption or decryption fails.
+     */
+    @RequestMapping(value = "v3/signed/generics", method = RequestMethod.POST)
+    @PowerAuth(resourceId = "/exchange/v3/signed/generics")
+    @PowerAuthEncryption(scope = EciesScope.ACTIVATION_SCOPE)
+    public ObjectResponse<DataExchangeResponse> exchangeSignedAndEncryptedDataGenerics(@EncryptedRequestBody ObjectRequest<DataExchangeRequest> request,
+                                                                                       EciesEncryptionContext eciesContext,
+                                                                                       PowerAuthApiAuthentication auth) throws PowerAuthAuthenticationException, PowerAuthEncryptionException {
+        if (auth == null || auth.getUserId() == null) {
+            logger.error("Signature validation failed");
```

Comment:
This should not be error, failed signature is a regular status of a user entering an incorrect PIN code.